tests/foreman/cli/test_host.py

```diff
@@ -70,1769 +64,1491 @@
-        cls.new_loc = make_location()
-        cls.new_lce = make_lifecycle_environment({'organization-id': cls.new_org['id']})
-        cls.LIBRARY = LifecycleEnvironment.info(
-            {'organization-id': cls.new_org['id'], 'name': ENVIRONMENT}
-        )
-        cls.DEFAULT_CV = ContentView.info(
-            {'organization-id': cls.new_org['id'], 'name': DEFAULT_CV}
-        )
-        cls.new_cv = make_content_view({'organization-id': cls.new_org['id']})
-        ContentView.publish({'id': cls.new_cv['id']})
-        version_id = ContentView.version_list({'content-view-id': cls.new_cv['id']})[0]['id']
-        ContentView.version_promote(
-            {
-                'id': version_id,
-                'to-lifecycle-environment-id': cls.new_lce['id'],
-                'organization-id': cls.new_org['id'],
-            }
-        )
-        cls.promoted_cv = cls.new_cv
-        # Setup for puppet class related tests
-        puppet_modules = [{'author': 'robottelo', 'name': 'generic_1'}]
-        cls.puppet_cv = publish_puppet_module(
-            puppet_modules, CUSTOM_PUPPET_REPO, cls.new_org['id']
-        )
-        cls.puppet_env = Environment.list(
-            {'search': 'content_view="{}"'.format(cls.puppet_cv['name'])}
-        )[0]
-        cls.puppet_class = Puppet.info(
-            {'name': puppet_modules[0]['name'], 'puppet-environment': cls.puppet_env['name']}
-        )
-        # adding org to a puppet env
-        Org.set_parameter(
-            {
-                'name': 'Environment',
-                'value': cls.puppet_env["name"],
-                'organization': cls.new_org["name"],
-            }
-        )
-
-    def setUp(self):
-        """Find an existing puppet proxy.
+@pytest.fixture(scope="module")
+def module_default_proxy():
+    """ Use the default installation smart proxy """
+    return Proxy.list({'search': f'url = https://{settings.server.hostname}:9090'})[0]
+
+
+@pytest.fixture(scope="function")
+def function_proxy():
+    proxy = make_proxy()
+    yield proxy
+    Proxy.delete({'id': proxy['id']})
+
+
+@pytest.fixture(scope="function")
+def function_host_content_source(
+    module_default_proxy, module_lce_library, module_org, module_published_cv
+):
+    host = make_fake_host(
+        {
+            'content-source-id': module_default_proxy['id'],
+            'content-view-id': module_published_cv.id,
+            'lifecycle-environment-id': module_lce_library.id,
+            'organization': module_org.name,
+        }
+    )
+    yield host
+    Host.delete({'id': host['id']})
+
+
+@pytest.fixture(scope="module")
+def module_make_ak(module_lce, module_org, module_promoted_cv):
+    return make_activation_key(
+        {
+            'content-view-id': module_promoted_cv.id,
+            'lifecycle-environment-id': module_lce.id,
+            'organization-id': module_org.id,
+        }
+    )
+
+
+@pytest.fixture(scope="function")
+def function_make_host(module_default_proxy):
```

Comment:
`s/make_//`?
\+ cleanup on teardown

Reply:
Unless you are making VMs there is no need for teardown, will just increase load on satellite under test